### DISCUSSION TITLE
How to resume async_pattern

### DISCUSSION THREAD

**Discussion by ibib** [snl-dakota/dakota#145] [Q&A] [answered]
Hello,
I am currently trying to do an async_pattern_search. 
Due to certain circumstances the evaluation has to run offline on another host. An *.out file is generated there. Now I have the problem how to guide this information back into Dakota. With centered_parameter_study I would just use the -post_run flag to read values from a tabular_data_file. But with async_pattern_search it's not supported. Is there some kind of workarround? 
Thank you!

**Comment by jadamstephens** [comment, marked as answer]
A limitation of Dakota is that we don't have good support for "offline" workflows like this. They are possible but require some workarounds. For a feedback style study, in which future evaluations depend on the results from earlier ones (such as an optimization), my experience is that using Dakota's [batch interface](https://snl-dakota.github.io/docs/latest_release/users/usingdakota/reference/interface-batch.html) is best. Dakota will write the parameters for all available evaluations to a single, concatenated parameters file, and it'll launch a single instance of your driver. It expects a single batch results file for all the evaluations in the batch upon driver exit. The batch interface is also described [here](https://snl-dakota.github.io/docs/latest_release/users/usingdakota/inputfile/interfaces/batchedevaluations.html).

To perform offline evaluation with the batch interface, initially set up your driver to exit without writing any results file. This will cause Dakota to abort. Run all the evaluations in the batch parameters file at your leisure and construct a batch results file. Start Dakota back up. It'll ingest the batch results file and write out the next batch of parameters. Rinse, repeat.

It's generally also necessary to use the [allow_existing_results](https://snl-dakota.github.io/docs/latest_release/users/usingdakota/reference/interface-analysis_drivers-fork-allow_existing_results.html) interface keyword to prevent Dakota from deleting the batch results file you  just created.
